<?xml version="1.0" encoding="UTF-8"?>
<definitions xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:flowable="http://flowable.org/bpmn" typeLanguage="http://www.w3.org/2001/XMLSchema"
             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
             xsi:schemaLocation="http://www.omg.org/spec/BPMN/20100524/MODEL http://www.omg.org/spec/BPMN/2.0/20100501/BPMN20.xsd"
             expressionLanguage="http://www.w3.org/1999/XPath" targetNamespace="http://www.flowable.org/test">
    <error id="myError" errorCode="myErrorCode1"/>
    <process id="processWithSingleExceptionMap" isExecutable="true">
        <startEvent id="theStart"/>
        <sequenceFlow id="flow1" sourceRef="theStart" targetRef="serviceTask"/>
        <serviceTask id="serviceTask" flowable:class="org.flowable.engine.test.bpmn.event.error.mapError.ThrowNestedCustomExceptionDelegate">
            <extensionElements>
                <flowable:mapException errorCode="myErrorCode1" rootCause="java.lang.IllegalArgumentException">org.flowable.engine.test.bpmn.event.error.mapError.BoundaryErrorParentException</flowable:mapException>
            </extensionElements>
        </serviceTask>
        <boundaryEvent id="catchError" attachedToRef="serviceTask">
            <extensionElements>
                <flowable:in target="errorMessageVar" source="errorMessage"/>
                <flowable:in target="errorCauseMessageVar" sourceExpression="${error.cause.message}"/>
            </extensionElements>
            <errorEventDefinition errorRef="myErrorCode1"/>
        </boundaryEvent>
        <sequenceFlow id="flow4" sourceRef="serviceTask" targetRef="theEnd"/>
        <endEvent id="theEnd"/>
        <serviceTask id="flagTask" name="Set Flag" flowable:class="org.flowable.engine.test.bpmn.event.error.mapError.FlagDelegate"/>
        <sequenceFlow id="flow5" sourceRef="catchError" targetRef="flagTask"/>
        <sequenceFlow id="flow6" sourceRef="flagTask" targetRef="theEnd"/>
    </process>
</definitions>